filegenerator/file_generator.go

```diff
@@ -0,0 +1,393 @@
+	allocps := allocpr.FindString(line)
+	bops := bopr.FindString(line)
+	mbpss := mbpsr.FindString(line)
+	var nsopi, allocpi, bopi int
+	var mbpsf float64
+	if nsops != "" {
+		nsopi, _ = strconv.Atoi(strings.Split(nsops, " ")[0])
+	}
+	if allocps != "" {
+		allocpi, _ = strconv.Atoi(strings.Split(allocps, " ")[0])
+	}
+	if bops != "" {
+		bopi, _ = strconv.Atoi(strings.Split(bops, " ")[0])
+	}
+	if mbpss != "" {
+		mbpsf, _ = strconv.ParseFloat(strings.Split(mbpss, " ")[0], 64)
+	}
+	return testName, BenchStats{N: nsopi, A: allocpi, B: bopi, M: mbpsf}
+}
+
+func isValidDataDir(dataDirName string) bool {
+	matched, _ := regexp.MatchString("[0-9][0-9]-[0-9][0-9]-[0-9][0-9][0-9][0-9]", dataDirName)
+	return matched
+}
+
+// RenderHistoricalBenchmarkResults takes the folders in benchSamples, parses them, and returns a
+// datastructure called BenchPackages which represent the data from the folders.
+func RenderHistoricalBenchmarkResults(dirs []string) BenchPackages {
+	bPackages := make(map[string]BenchTestMap)
+	benchSamples := os.Getenv(benchSamplesEnv)
+	dataDirs, err := ioutil.ReadDir(benchSamples)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, dataDir := range dataDirs {
+		if !dataDir.IsDir() {
+			continue
+		}
+		dataDirName := dataDir.Name()
+		if !isValidDataDir(dataDirName) {
+			continue
+		}
+		for _, dirName := range dirs {
+			if _, ok := bPackages[dirName]; !ok {
+				bPackages[dirName] = make(BenchTestMap)
+			}
+			fullDirPath := benchSamples + "/" + dataDirName + "/cockroach/" + dirName
+			files, err := ioutil.ReadDir(fullDirPath)
+			if err != nil {
+				log.Fatal(err)
+			}
+			for _, file := range files {
+				fileName := file.Name()
+				matched, _ := regexp.MatchString(".*test\\.stdout", fileName)
+				if matched {
+					content, err := ioutil.ReadFile(fullDirPath + "/" + fileName)
+					if err != nil {
+						log.Fatal(err)
+					}
+					for _, line := range strings.Split(string(content), "\n") {
+						if isValidTestLine(line) {
+							testName, stats := parseTestResultLine(line)
+							if _, ok := bPackages[dirName][testName]; !ok {
+								bPackages[dirName][testName] = make(BenchResults)
+							}
+							bPackages[dirName][testName][dataDirName] = stats
+						}
+					}
+				}
+			}
+		}
+	}
+	return bPackages
+}
+
+// GenerateJSONFiles takes a BenchPackages and creates a json file for every test
+// containing its benchmark results data over time.
+func GenerateJSONFiles(packages BenchPackages) {
+	deployRoot := os.Getenv(deployRootEnv)
+	for dir := range packages {
+		for testName := range packages[dir] {
+			testData := packages[dir][testName]
+			fileFullName := deployRoot + "/" + dir + "/" + testName + ".json"
+			testDataJSON, err := json.Marshal(&testData)
+			if err != nil {
+				log.Fatal(err)
+			}
+			err = ioutil.WriteFile(fileFullName, testDataJSON, 0644)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+}
+
+// GenerateTestNameJSONFile Generates a file that maps directory to testnames,
```

Comment:
I'm still not sure what this function does even after reading this comment. Try to rephrase.